tests/logging/test_celery.py

```diff
@@ -0,0 +1,106 @@
+import json
+import os
+import subprocess
+import warnings
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from notifications_utils.logging.celery import set_up_logging, setup_logging_connect
+
+
+def test_set_up_logging_success():
+    """Test that set_up_logging correctly overrides warnings.showwarning."""
+    logger = MagicMock()
+    set_up_logging(logger)
+
+    # Assert that warnings.showwarning is overridden
+    assert warnings.showwarning != warnings._showwarning_orig
+
+    # Simulate a warning and check if it logs correctly
+    warnings.showwarning("Test message", UserWarning, "test_file.py", 42)
+    logger.warning.assert_called_once_with(
+        "Test message",
+        {
+            "level": "WARNING",
+            "message": "Test message",
+            "category": "UserWarning",
+            "filename": "test_file.py",
+            "lineno": 42,
+        },
+    )
+
+
+def test_set_up_logging_missing_logger():
+    logger = None
+
+    with pytest.raises(AttributeError, match="The provided logger object is invalid."):
+        set_up_logging(logger)
+
+
+@patch("notifications_utils.logging.celery.dictConfig")
+def test_setup_logging_connect_success(mock_dict_config):
+    """Test that setup_logging_connect successfully configures logging."""
+    setup_logging_connect()
+
+    # Assert that dictConfig was called
+    mock_dict_config.assert_called_once()
+
+
+def test_celery_mock_logs():
+    command = [
+        "celery",
+        "--quiet",
+        "-A",
+        "celery_mock",
+        "worker",
+        "--loglevel=INFO",
+    ]
+
+    # Set the environment variable
+    env = os.environ.copy()
+    env["CELERY_LOG_LEVEL"] = "INFO"
+
+    # Change the working directory to the mock folder
+    mock_dir = os.path.join(os.path.dirname(__file__), "mock")
+
+    try:
+        # Start the Celery worker process
+        process = subprocess.Popen(
+            command,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+            env=env,
+            cwd=mock_dir,
+        )
+
+        process.wait(timeout=10)
```

Comment:
Why do we call `wait` before `communicate` here?

Just wondering what happens if the buffer for stdout or stderr fills up here, presumably we get a timeout error?

Reply:
You are right. Not ideal. Updated.